src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -68,27 +65,29 @@ export interface IdentityProviderHttpHandlerArgs extends BaseHttpHandlerArgs {
  * This handler handles all requests since it assumes all those requests are relevant for the IDP interaction.
  * A {@link RouterHandler} should be used to filter out other requests.
  */
-export class IdentityProviderHttpHandler extends BaseHttpHandler {
+export class IdentityProviderHttpHandler extends OperationHttpHandler {
   protected readonly logger = getLoggerFor(this);
 
   private readonly baseUrl: string;
   private readonly providerFactory: ProviderFactory;
   private readonly interactionRoutes: InteractionRoute[];
   private readonly converter: RepresentationConverter;
   private readonly interactionCompleter: InteractionCompleter;
+  private readonly errorHandler: ErrorHandler;
```

Comment:
On base?

Reply:
The IDP also needs an error handler, it uses it to convert errors thrown by the handlers to a partial template so it can inject them in the page the request was sent from. So you see "missing email" on the register page instead of being sent to a new error page with that information.

Reply:
I now wonder to what extent this can be performed by another handler that is next in the chain.
But the problem there is probably how to pass that error.

And this is not something we want on the base class?

Reply:
The base class handles errors, in the sense that it will return an error description with the correct content type. But the IDP wants to return the original page with the error embedded in it.

In my opinion, the correct way to handle this (and a few other IDP routing issues) is to move part of the behavior to actual IDP HTML pages. These should use javascript to call the server and handle the embedding of the error in the page (and also prefilled data for example) themselves. But that requires a refactor there first. Part of the behavior of the IDP handler is is simply there to support our current versions of those pages.